src/Microsoft.Health.Fhir.Core/Exceptions/FhirServiceUnavailableException.cs

```diff
@@ -0,0 +1,25 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Exceptions
+{
+    public sealed class FhirServiceUnavailableException : FhirException
```

Comment:
Could we reuse `Microsoft.Health.Abstractions.Exceptions.ServiceUnavailableException`?

Reply:
Unfortunately, this class does not have good constructors: it does not let provide an inner exception and an additional error message.

Reply:
Ah I see, but also a fixable problem ;)